infra/general/languageCodeManager.php

```diff
@@ -0,0 +1,316 @@
+        self::addLanguageToArrays($tmpArrTwoCode,$tmpArrThreeCodeT,$tmpArrKalturaName,'VI', "vi", "vie", "vie", "Vietnamese", "Ti\xe1\xba\xbfng Vi\xe1\xbb\x87t",'Vietnamese');
+        self::addLanguageToArrays($tmpArrTwoCode,$tmpArrThreeCodeT,$tmpArrKalturaName,'VO', "vo", "vol", "vol", "Volap\xc3\xbck", "Volap\xc3\xbck",'Volapuk');
+        self::addLanguageToArrays($tmpArrTwoCode,$tmpArrThreeCodeT,$tmpArrKalturaName,'WA', "wa", "wln", "wln", "Walloon", "walon",'Walloon');//not yet defined in kaltura
+        self::addLanguageToArrays($tmpArrTwoCode,$tmpArrThreeCodeT,$tmpArrKalturaName,'CY', "cy", "cym", "wel", "Welsh", "Cymraeg",'Welsh');
+        self::addLanguageToArrays($tmpArrTwoCode,$tmpArrThreeCodeT,$tmpArrKalturaName,'WO', "wo", "wol", "wol", "Wolof", "Wollof",'Wolof');
+        self::addLanguageToArrays($tmpArrTwoCode,$tmpArrThreeCodeT,$tmpArrKalturaName,'FY', "fy", "fry", "fry", "Western Frisian", "Frysk",'Frisian');
+        self::addLanguageToArrays($tmpArrTwoCode,$tmpArrThreeCodeT,$tmpArrKalturaName,'XH', "xh", "xho", "xho", "Xhosa", "isiXhosa",'Xhosa');
+        self::addLanguageToArrays($tmpArrTwoCode,$tmpArrThreeCodeT,$tmpArrKalturaName,'YI', "yi", "yid", "yid", "Yiddish", "\xd7\x99\xd7\x99\xd6\xb4\xd7\x93\xd7\x99\xd7\xa9",'Yiddish');
+        self::addLanguageToArrays($tmpArrTwoCode,$tmpArrThreeCodeT,$tmpArrKalturaName,'YO', "yo", "yor", "yor", "Yoruba", "Yor\xc3\xb9",'Yoruba');
+        self::addLanguageToArrays($tmpArrTwoCode,$tmpArrThreeCodeT,$tmpArrKalturaName,'ZA', "za", "zha", "zha", "Zhuang, Chuang", "Sa\xc9\xaf cue\xc5\x8b\xc6\x85, Saw cuengh",'Zhuang, Chuang');//not yet defined in kaltura
+        self::addLanguageToArrays($tmpArrTwoCode,$tmpArrThreeCodeT,$tmpArrKalturaName,'ZU', "zu", "zul", "zul", "Zulu", "isiZulu",'Zulu');
+
+        //The following cases are not part of language ISO , they are added to support backward compatibility
+        self::addLanguageToArrays($tmpArrTwoCode,$tmpArrThreeCodeT,$tmpArrKalturaName,'SH','sh','shc','shc','Serbo-Croatian','Serbo-Croatian' ,'Serbo-Croatian');
+        self::addLanguageToArrays($tmpArrTwoCode,$tmpArrThreeCodeT,$tmpArrKalturaName,'EN_GB', "en", "enb", "enb", "English (British)", "English (British)",'English (British)');
+        self::addLanguageToArrays($tmpArrTwoCode,$tmpArrThreeCodeT,$tmpArrKalturaName,'EN_US', "en", "enu", "enu", "English (American)", "English (American)",'English (American)');
+        self::addLanguageToArrays($tmpArrTwoCode,$tmpArrThreeCodeT,$tmpArrKalturaName,'MO','mo','mol','mol','Moldavian','Moldavian','Moldavian');
+        self::addLanguageToArrays($tmpArrTwoCode,$tmpArrThreeCodeT,$tmpArrKalturaName,'MU','mu','mul','mul','Multilingual','Multilingual','Multilingual');
+        self::addLanguageToArrays($tmpArrTwoCode,$tmpArrThreeCodeT,$tmpArrKalturaName,'UN', "un", "und", "und", "Undefined", "Undefined","Undefined");
+
+        $strArr = var_export($tmpArrTwoCode,true);

```

Comment:
Create a function that gets array+name and returns a buffer and use it in the 3 cases